src/models/IPREventTypes.ts

```diff
@@ -0,0 +1,26 @@
+export interface IpvStartedOnEvent {
+	":ipvStartedOn": string;
+	":userEmail": string;
+	":nameParts": Array<{
+		type: string;
+		value: string;
+	}>;
+	":clientName": string;
+	":redirectUri": string;
+}
+
+export interface JourneyWentAsyncOnEvent {
+	":journeyWentAsyncOn": string;
+}
+
+export interface ReadyToResumeOnEvent {
+	":readyToResumeOn": string;
+}
+
+export interface AccountDeletedOnEvent {
```

Comment:
As discussed on call yesterday, we need the data model to reflect {
  userEmail [string],
  nameParts [Object]
  clientName [string],
  redirectUri [string],
  
  userid [string],
  
  ipvStartedOn [timestamp],
  journeyWentAsyncOn [timestamp],
  readyToResumeOn [timestamp],
  accountDeletedOn [timestamp],
  
  ttl/expiresOn [timestamp]
}

Reply:
Could you point out where we would be referencing that data model in the code please

Reply:
So I had a crack at this, I think the idea of having a datamodel makes sense [here](https://github.com/alphagov/di-ipv-cri-cic-api/blob/main/src/services/SessionRequestProcessor.ts#L137) because the data model we pass to the call to dynamodb never changes however in this file we have different datasets we save to the DB depending on which event we receive so I'm really struggling to find where and why a data model would be of any use.. The only thing all 4 types of events share in common is expiry date all other fields are different based on what event we receive so we'd basically end up with 4 data models